### ISSUE TITLE
ComboBox: provide methods to control popup list

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#366]
There's currently no way to programmatically hide the ComboBox's list.  This would be useful in situations like https://github.com/phetsims/sun/issues/237 and https://github.com/phetsims/projectile-motion/issues/164.  When the browser window is resize, ScreenView.layout is called.  Subtypes of ScreenView would override layout to handle hiding the ComboBox list.  Something like:

```js
// @override
layout: function( width, height ) {
  this.myComboBox.hideList();
  ScreenView.prototype.layout.call( this, width, height );
}
```

Should investigate whether `hideList` is sufficient, or whether it would also be useful to have `showList` and `isListVisible`.

**Comment by pixelzoom** [comment]
@ariel-phet If there is consensus that this would be a useful enhancement, consider having @andrea-phet implement, since she could use it immediately in https://github.com/phetsims/projectile-motion/issues/164.

**Comment by andrealin** [comment]
I added `showList` and `hideList` to ComboBox.

There are some checks within showList and hideList about the listNode visibility, so would `isListVisible` still be necessary?

**Comment by andrealin** [comment]
@pixelzoom could you take a look at the new functionality added and let me know if changes/improvements are needed? See [above comment](https://github.com/phetsims/sun/issues/366#issuecomment-399158040)

**Comment by pixelzoom** [comment]
Too much of the implementation for `showList` and `hideList` was in the constructor.  In this case, I think it's making it more difficult to read and (of secondary concern) the implementation of the functions is duplicated in every instance of ComboBox.   So I created some addition fields (`this.listNode`, etc.) and moved the full implementations of `showList`, `histList` and `moveList` to the prototype.  

I tested with a few sims that use ComboBox (including projectile-motion) and I didn't see any problems.

@andrealin Back to you to review.  After a few rounds of CT, we can close this issue.
